src/plugins/discourse/createGraph.test.js

```diff
@@ -0,0 +1,317 @@
+// @flow
+
+import sortBy from "lodash.sortby";
+import type {DiscourseData} from "./mirror";
+import type {Topic, Post, PostId, TopicId} from "./fetch";
+import {NodeAddress, EdgeAddress, type Node, type Edge} from "../../core/graph";
+import {
+  createGraph,
+  userNode,
+  topicNode,
+  postNode,
+  authorsTopicEdge,
+  authorsPostEdge,
+  topicContainsPostEdge,
+  postRepliesEdge,
+} from "./createGraph";
+import {
+  userNodeType,
+  topicNodeType,
+  postNodeType,
+  authorsTopicEdgeType,
+  authorsPostEdgeType,
+  topicContainsPostEdgeType,
+  postRepliesEdgeType,
+} from "./declaration";
+import type {EdgeType, NodeType} from "../../analysis/types";
+
+describe("plugins/discourse/createGraph", () => {
+  class MockData implements DiscourseData {
+    _topics: $ReadOnlyArray<Topic>;
+    _posts: $ReadOnlyArray<Post>;
+
+    constructor(topics, posts) {
+      this._topics = topics;
+      this._posts = posts;
+    }
+    topics(): $ReadOnlyArray<Topic> {
+      return this._topics;
+    }
+    posts(): $ReadOnlyArray<Post> {
+      return this._posts;
+    }
+    users(): $ReadOnlyArray<string> {
+      const users = new Set();
+      for (const {authorUsername} of this.posts()) {
+        users.add(authorUsername);
+      }
+      for (const {authorUsername} of this.topics()) {
+        users.add(authorUsername);
+      }
+      return Array.from(users);
+    }
+    findPostInTopic(topicId: TopicId, indexWithinTopic: number): ?PostId {
+      const post = this._posts.filter(
+        (p) => p.topicId === topicId && p.indexWithinTopic === indexWithinTopic
+      )[0];
+      return post ? post.id : null;
+    }
+  }
+
+  function example() {
+    const url = "https://url.com";
```

Comment:
nit: It’s conventional to use <https://example.com> as an example URL,
because it’s owned by ICANN and explicitly allowed to be used in such
cases. By contrast, <https://url.com> is owned by some company.